tradebot/config/manager.py

```
@@ -0,0 +1,222 @@
+            with open(self.config_path, 'r') as f:
+                config = yaml.safe_load(f)
+            
+            if config is None:
+                config = {}
+            
+            logger.info(f"Configuration loaded from {self.config_path}")
+            return config
+            
+        except yaml.YAMLError as e:
+            from tradebot.exceptions import ConfigurationError
+            raise ConfigurationError(f"Invalid YAML format in {self.config_path}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)